agent/agent.go

```diff
@@ -873,31 +895,6 @@ func (a *Agent) handleTimeoutAndOOM(ctx context.Context, tc *taskContext, status
 	}
 }
 
-func (a *Agent) runTaskTimeoutCommands(ctx context.Context, tc *taskContext) {
-	tc.logger.Task().Info("Running task-timeout commands.")
-	start := time.Now()
-
-	timeout, err := tc.getTimeout(tc.taskGroup)
-	if err != nil {
-		tc.logger.Execution().Error(errors.Wrap(err, "fetching task group for task timeout commands"))
-		return
-	}
-	if timeout.commands != nil {
-		timeoutCtx, timeoutCancel := context.WithCancel(ctx)
-		defer timeoutCancel()
-		timeoutOpts := timeoutWatcherOptions{
-			tc:         tc,
-			kind:       timeout.timeoutKind,
-			getTimeout: timeout.getTimeout,
-		}
-		go a.startTimeoutWatcher(timeoutCtx, timeoutCancel, timeoutOpts)
-
-		err := a.runCommandsInBlock(timeoutCtx, tc, timeout.commands.List(), runCommandsOptions{block: timeout.block})
-		tc.logger.Task().Error(errors.Wrap(err, "Running timeout commands failed"))
```

Comment:
Why was this line removed and the error is ignored now? Same question for `runEndTaskSync` etc

Reply:
Hm, so [the command failure already gets logged further down](https://github.com/evergreen-ci/evergreen/blob/a2958b866cf95565441d94b807834220c8a45449/agent/command.go#L294) in the command logic, so the user will still see that a command failed. From what I can see, the `"Running <block name> commands failed"` is specifically used in cases where the command both failed _and_ that failure will cause the whole task to fail (which will also stop regular command execution and skip to `post`).

In the case of `timeout`, the agent is already going to mark the task as failed since it timed out, so `timeout` can't change an already-failed task. For task sync, they're considered "optional" commands because they don't influence the task status.

Reply:
Got it, that makes sense, thanks